android/app/src/main/kotlin/com/therealansh/iiit_sonepat_stable/MainActivity.kt

```diff
@@ -1,6 +1,128 @@
 package com.therealansh.iiit_sonepat_stable
 
+
+import android.content.Context
+import android.content.ContextWrapper
+import android.content.Intent
+import android.content.IntentFilter
+import android.os.Build.VERSION
+import android.os.Build.VERSION_CODES
+import androidx.annotation.NonNull
 import io.flutter.embedding.android.FlutterActivity
+import io.flutter.embedding.engine.FlutterEngine
+import io.flutter.plugin.common.MethodChannel
+import android.content.BroadcastReceiver
+import android.os.Bundle
+import android.view.View
+import android.widget.EditText
+import androidx.appcompat.app.AppCompatActivity
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import org.jitsi.meet.sdk.*
+import timber.log.Timber
+import java.net.MalformedURLException
+import java.net.URL
+
 
 class MainActivity: FlutterActivity() {
+
+    private val CHANNEL = "samples.flutter.dev/meet" //for Server Side of Kotlin platform
+
+    private val broadcastReceiver = object : BroadcastReceiver() {
+        override fun onReceive(context: Context?, intent: Intent?) {
+            onBroadcastReceived(intent)
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // Initialize default options for Jitsi Meet conferences.
+        val serverURL: URL = try {
+            // When using JaaS, replace "https://meet.jit.si" with the proper serverURL
+            URL("https://meet.jit.si")
+        } catch (e: MalformedURLException) {
+            e.printStackTrace()
+            throw RuntimeException("Invalid server URL!")
+        }
+        val defaultOptions = JitsiMeetConferenceOptions.Builder()
+                .setServerURL(serverURL)
+                // When using JaaS, set the obtained JWT here
+                //.setToken("MyJWT")
+                // Different features flags can be set
+                //.setFeatureFlag("toolbox.enabled", false)
+                //.setFeatureFlag("filmstrip.enabled", false)
+                .setWelcomePageEnabled(false)
+                .build()
+        JitsiMeet.setDefaultConferenceOptions(defaultOptions)
+
+        registerForBroadcastMessages()
+    }
+
+    override fun onDestroy() {
+        LocalBroadcastManager.getInstance(this).unregisterReceiver(broadcastReceiver)
+        super.onDestroy()
+    }
+
+    var usertext:String?=""//For storing UserName
+
+    override fun configureFlutterEngine(@NonNull flutterEngine: FlutterEngine) {
+        super.configureFlutterEngine(flutterEngine)
+        MethodChannel(flutterEngine.dartExecutor.binaryMessenger, CHANNEL).setMethodCallHandler { call, result ->
+
+            if(call.method == "Meeting started")
+                usertext=call.argument<String>("text")
+            onButtonClick(usertext)
+
+        }
+    }
+
+    fun onButtonClick(text:String?) {
```

Comment:
we can handle error here too, we could raise an Exception here in case a user is unable to join the call. This exception can be showed at the frontend. Read about ``` try{} catch(err) {}``` block statement to raise an exception and we could show a dialog box in the frontend if we catch an error